src/screens/Followings/elements/RenderList/index.js

```diff
@@ -0,0 +1,157 @@
+    alignItems: 'center',
+    paddingVertical: 5
+  },
+  imageProfile: {
+    width: 48,
+    height: 48,
+    borderRadius: 48
+  },
+  wrapTextProfile: {
+    flex: 1,
+    flexDirection: 'column',
+    justifyContent: 'space-between',
+    marginLeft: 12
+  },
+  textProfileUsername: {
+    fontFamily: fonts.inter[500],
+    fontWeight: 'bold',
+    fontSize: 14,
+    color: colors.black,
+    lineHeight: 16.94
+  },
+  textProfileFullName: {
+    fontSize: 12,
+    color: colors.gray,
+    fontFamily: fonts.inter[400],
+    lineHeight: 18,
+    flexWrap: 'wrap'
+  },
+  buttonFollowing: {
+    width: 88,
+    height: 36,
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'center',
+    borderColor: colors.bondi_blue,
+    borderWidth: 1,
+    borderRadius: 8
+  },
+  card: {
+    height: 68,
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    width: '100%',
+    marginVertical: 10
+  }
+});
+
+const DomainList = (props) => {
+  const {item, isHashtag, handleSetFollow, handleSetUnFollow, onPressBody} = props;
+
+  return (
+    <View style={styles.card}>
+      <GlobalButton onPress={() => onPressBody(item)} buttonStyle={styles.wrapProfile}>
+        {!isHashtag ? (
+          <React.Fragment>
+            {item.image ? (
+              <Image
+                source={{
+                  uri: item.image
+                }}
+                style={styles.profilepicture}
+                width={48}
+                height={48}
+              />
+            ) : (
+              <View style={styles.profilepicture} />
+            )}
+          </React.Fragment>
+        ) : null}
+
+        <View style={styles.wrapTextProfile}>
+          <Text style={styles.textProfileUsername}>
+            {isHashtag && '#'}
+            {item.name}
+          </Text>
+          <Text style={styles.textProfileFullName} numberOfLines={1} ellipsizeMode={'tail'}>
+            {item.description ? item.description : ''}
+          </Text>
+        </View>
+      </GlobalButton>
+      {item.isunfollowed ? (
+        <GlobalButton onPress={handleSetFollow}>
+          <View style={styles.buttonFollow}>
+            <Text style={styles.textButtonFollow}>{isHashtag ? 'Join' : 'Follow'}</Text>
+          </View>
+        </GlobalButton>
+      ) : (
+        <GlobalButton onPress={handleSetUnFollow}>
+          <View style={styles.buttonFollowing}>
+            <Text style={styles.textButtonFollowing}>{isHashtag ? 'Joined' : 'Following'}</Text>
+          </View>
+        </GlobalButton>
+      )}
+    </View>
+  );
+};
+
+DomainList.propTypes = {
+  item: PropTypes.object,
+  isHashtag: PropTypes.bool,
+  onPressBody: PropTypes.func,
+  handleSetFollow: PropTypes.func,
+  handleSetUnFollow: PropTypes.func
+};
+
+DomainList.defaultProps = {
+  handleSetFollow: () => null,
+  handleSetUnFollow: () => null,
+  onPressBody: () => null
+};
+
+export default React.memo(DomainList);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `DomainList` component seems to be well implemented. However, it's important to note that the prop types defined for this component are quite loose. For instance, `item` is defined as an object but there's no specification of what properties this object should have. This could lead to potential issues if the object doesn't have the expected properties. It would be better to use `PropTypes.shape` to specify the structure of the `item` object.

```diff
-  item: PropTypes.object,
+  item: PropTypes.shape({
+    image: PropTypes.string,
+    name: PropTypes.string,
+    description: PropTypes.string,
+    isunfollowed: PropTypes.bool,
+  }),
```

Also, the default props for the functions `handleSetFollow`, `handleSetUnFollow`, and `onPressBody` are set to `() => null`. It would be more appropriate to use `() => {}` as a default function since it does not imply any return value.

```diff
-  handleSetFollow: () => null,
-  handleSetUnFollow: () => null,
-  onPressBody: () => null
+  handleSetFollow: () => {},
+  handleSetUnFollow: () => {},
+  onPressBody: () => {}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->